### ISSUE TITLE
BUG: MNE_BROWSER_PRECOMPUTE=true broken

### ISSUE THREAD

**Issue by larsoner** [mne-tools/mne-qt-browser#270]
On my system I have a lot more memory compared to size of files that I open, so for almost all files precomputation runs.

In theory this is a good thing, but in practice I've noticed some weird bugs where the display doesn't always update properly when I click and unclick bad channels, or enable/disable projectors.

I think relatedly, in #268 the updates only happened properly when I had `MNE_BROWSER_PRECOMPUTE=false` set. Otherwise they didn't take effect. Interestingly, the traces *were* updated when hitting plus or minus on the keyboard. This is maybe some hint for how things should be changed/fixed.

But in any case, we should probably do a comprehensive check of visual output with precompute mode on and off, for at least the following behaviors:

1. Enabling/disabling projectors
2. Clicking/unclicking bad channels that are in projectors when projection is on (as this changes the projector that is applied)
3. Changing channel scalings using plus/minus on the keyboard
4. Changing channel scalings using the new channel-specific UI

@nmarkowitz for now you don't need to worry about any of this stuff -- I just need to remember to test with `MNE_BROWSER_PRECOMPUTE=false` (which I'll do by setting my system config value now).

**Comment by larsoner** [comment]
Whitened channels and maybe stim channels are scaled differently in `_process_data` in MNE, too, so we need to make sure we treat them properly here.

**Comment by larsoner** [comment]
Wow that's quite wrong! 😱 

Yes we should try to fix this precompute issue soon if possible

**Comment by marsipu** [comment]
@larsoner I noticed, that when precompute=True on selecting bad-channels, each time _apply_update_projectors is called and the data is precomputed again. Is that expected or is it related to this issue?

**Comment by larsoner** [comment]
Yep, if the bad channel selection changes the projector should be updated and reapplied!
